nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -72,7 +72,16 @@ StaticLayer::onInitialize()
   global_frame_ = layered_costmap_->getGlobalFrameID();
 
   getParameters();
-  getMap();
+
+  if (!first_map_from_service_ && first_map_only_) {
```

Comment:
I recommend moving this semantic check for parameter values into the getParameters function. In other modules, when there are enough parameters to warrant a separate function, like getParameters, I've tried to consistently use the pattern of first getting the parameters and then performing any required semantic checks on the values.

Reply:
Done.